docs/concept/property-repair.md

```diff
@@ -0,0 +1,142 @@
+   * If the maximum count reaches zero, the process is considered complete, and it proceeds to step 7.
+   * Otherwise, the message is forwarded to the next node, repeating steps 3–6.
+7. **Send Result to Origin Node**: The current node sends the result—either success or failure—to the origin node specified within the context.
+8. **Origin Node Receives the Result**: The initiating sender node receives the final outcome of the gossip protocol and can proceed with any post-processing logic.
+
+### Example of Message Propagation
+
+To illustrate the gossip message propagation process, consider a scenario with three nodes with the version: A(version 2), B(version 1), C(version 3), and the sender is node B. The sequence of operations is as follows:
+
+1. B node building the context, and the maximum count is calculated as `(3) * 2 - 3 = 3`.
+2. B node sends the gossip message to A as its first node in the nodes list.
+3. A node receives the message, identifies B as the next node, and do peer-to-peer interaction, current nodes and the version is: A(version 2), B(version 2), C(version 3)
+4. A node sends the gossip message to B, then B node receives the message, identifies C as the next node, and do peer-to-peer interaction, current nodes and the version is: A(version2), B(version 3), C(version 3)
+5. B node sends the gossip message to C, then C node receives the message, identifies A as the next node, and do peer-to-peer interaction, current nodes and the version is: A(version 3), B(version 3), C(version 3)
+6. finally, the maximum count decremented to the zero, means the gossip propagation is completed, and C node sends the result the original node B.
+7. B node received the result, and the gossip propagation is completed.
+
+### Tracing
+
+Tracing becomes critically important in a gossip-based protocol, as issues can arise at any stage of the communication and synchronization process.
+New trace spans are created in the following scenarios:
+
+1. **Initiating Node**: The initiating node records the full trace of the entire request, capturing all steps from message dispatch to the final result collection.
+2. **Receiving Sync Requests**: When a node receives a sync request and begins communication with another peer node, it creates a new trace span for that interaction to track the request lifecycle independently.
+3. **Business Logic Execution**: During the actual business processing (e.g., data comparison, update, and transmission), custom trace data and metadata can be attached to the active trace context to provide deeper insights into logic-specific behaviors.
+
+After each synchronization cycle, the receiving node will send its trace data back to the initiating node, allowing the initiator to aggregate and correlate all spans for end-to-end visibility and debugging.
+
+## Property Repair
+
+Once all the above preparations are complete, the system can proceed with the Property Repair process.
```

Comment:
What's the "preparations"? The previous section is "Gossip Protocol," which is not an action.